File: prince-of-versions/src/main/java/co/infinum/princeofversions/StreamLoader.java
package co.infinum.princeofversions;

import java.io.IOException;
import java.io.InputStream;

/**
 * Represents a concrete loader that load resource from stream.
 */
public final class StreamLoader implements Loader {

    /**
     * Used input stream.
     */
    private InputStream is;

    /**
     * Creates a new stream loader using provided input stream.
     *
     * @param is Input stream.
     */
    public StreamLoader(InputStream is) {
        this.is = is;
    }

    @Override
    public String load() throws IOException {
        return StreamIo.toString(is);
    }
}

File: prince-of-versions/src/main/java/co/infinum/princeofversions/StreamLoader.kt
package co.infinum.princeofversions

import java.io.IOException
import java.io.InputStream

/**
 * Represents a concrete loader that loads a resource from an [InputStream].
 *
 * @param inputStream The input stream to load the resource from.
 */
class StreamLoader(private val inputStream: InputStream) : Loader {

    @Throws(IOException::class)
    override fun load(): String = StreamIo.toString(inputStream)
}
